src/hotspot/share/gc/shenandoah/shenandoahUncommitThread.hpp

```diff
@@ -66,6 +60,8 @@ class ShenandoahUncommitThread : public ConcurrentGCThread {
   // True if the control thread has allowed this thread to uncommit regions
   bool is_uncommit_allowed() const;
 
+  size_t do_uncommit_work(double shrink_before, size_t shrink_until) const;
```

Comment:
A comment describing arguments and return value would be nice.  Mention premature termination if gc is requested.